frontend/src/App.js

```diff
@@ -1,55 +1,115 @@
+      const chunk = arr.splice(0, chunkSize);
+      res.push(chunk);
   }
+  return res;
 }
 
-const matchup1 = new Matchup('USA', 'Mexico')
-const matchup2 = new Matchup('Spain', 'Portugal')
-const matchup3 = new Matchup('Canada', 'Honduras')
 
-function App() {
+// Return an array containg sub-arrays of 6 matches
+function getChunkedGroupStageMatches(matches) {
+  const groupStageMatches = matches.filter(
+    match => match.comp_stage === "Group Stage");
+  const matchupChunks = spliceIntoChunks(groupStageMatches, 6)
+  return matchupChunks
+ }
+
+
+class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      matches: [],
+      teams: []
+    }
+    
+  }
+
+  // Make ALL API calls here so that all future munging is in-memory
+  async componentDidMount() {
 
-  const renderMatchupRow = (matchup) => {
+    const matches = await callApi('matches')
+    const teams = await callApi('teams')
+    
+    var teamNamesById = {}
+    teams.forEach((team) => {
+      teamNamesById[team.id] = team.name
+    })
+    this.setState({matches: matches, teamNamesById: teamNamesById})
+  }
+
+  renderMatchupRow(matchup, index) {
+    const homeTeamName = this.state.teamNamesById[matchup.home_team]
+    const awayTeamName = this.state.teamNamesById[matchup.away_team]
+    const vsText = this.state.matches ? 'vs' : ''
     return (
-      <tr className="card-table-row">
-        <td>{matchup.team1}</td>
-        <td className="vs-cell">vs</td>
-        <td>{matchup.team2}</td>
+      <tr key={index.toString()} className="card-table-row">
+        <td>{homeTeamName}</td>
+        <td className="vs-cell">{vsText}</td>
+        <td>{awayTeamName}</td>
       </tr>
     )
   }
 
-  const renderCard = (groupNum, matchup1, matchup2, matchup3) => {
+  renderMatchupRows(matches) {
+    const matchupRows = []
+    matches.forEach((match, index) => {
+      const matchupRow = this.renderMatchupRow(match, index)
+      matchupRows.push(matchupRow)
+    })
+    return matchupRows
+  }
+
+  renderCard(groupNum, matches) {
+    const matchupRows = this.renderMatchupRows(matches)
     return (
       <div className="group-card">
         <h2 className="group-num">Group {groupNum}</h2>
         <table className="card-table">
           <tbody>
-            {renderMatchupRow(matchup1)}
-            {renderMatchupRow(matchup2)}
-            {renderMatchupRow(matchup3)}
+            {matchupRows}
           </tbody>
         </table>
       </div>
     )
   }
 
-  return (
-    <div className="App">
-      <h1 id="main-header">World Cup 2018 Analytics</h1> 
-      <div className="group-scroller-container">
-        {renderCard(1, matchup1, matchup2, matchup3)}
-        {renderCard(2, matchup1, matchup2, matchup3)}
-        {renderCard(3, matchup1, matchup2, matchup3)}
-        {renderCard(4, matchup1, matchup2, matchup3)}
+  renderCards() {
+    const chunkedGSMatches = getChunkedGroupStageMatches(this.state.matches)
```

Comment:
I had this in `this.state` in a previous revision, but I think it's cleaner to remove it since it just duplicates a lot of other information that's already in `this.state`